hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPOfferService.java

```diff
@@ -679,16 +679,19 @@ boolean processCommandFromActor(DatanodeCommand cmd,
       actor.reRegister();
       return false;
     }
+    boolean isActiveActor;
     writeLock();
     try {
-      if (actor == bpServiceToActive) {
-        return processCommandFromActive(cmd, actor);
-      } else {
-        return processCommandFromStandby(cmd, actor);
-      }
+      isActiveActor = actor == bpServiceToActive;
     } finally {
       writeUnlock();
     }
+
+    if (actor == bpServiceToActive) {
+      return processCommandFromActive(cmd, actor);
+    } else {
+      return processCommandFromStandby(cmd, actor);
+    }
```

Comment:
Even before HDFS-6788, this part was at least covered by synchronized lock on the actor thread

Reply:
Hi, @virajjasani . thanks for your careful review.  Surely, before [HDFS-6788](https://issues.apache.org/jira/browse/HDFS-6788)， this part was covered by synchronized lock.
but in method `processCommandFromActive` and `processCommandFromStandby`,  it just use the parameter actor to print log info. The lock here is just trying to decide actor is whether bpServiceToActive or not and determine to execute either processCommandFromActive or processCommandFromStandby.  

when occurs switchover between active namenode and standby namenode,  the datanodes would be set to stale status, in stale status, we are not allowed to delete blocks directly, we put those blocks into postponedMisreplicatedBlocks.  So, even we execute the DatanodeCommand from the previous active namenode(now standby), it is okay.